selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/user/TestUserImpl.java

```diff
@@ -47,12 +48,12 @@
   /** To instantiate user with generated email and password. */
   @Inject
   public TestUserImpl(
-      TestUserServiceClient userServiceClient,
+      Provider<TestUserServiceClient> userServiceClient,
       TestAuthServiceClient authServiceClient,
       TestWorkspaceServiceClientFactory wsServiceClientFactory)
       throws Exception {
     this(
-        userServiceClient,
+        userServiceClient.get(),
```

Comment:
You might pass `null` ^(

Reply:
In which sittuation null could be there? There is **@Inject** annotation above the constructor, and **Guice** will throw an error instantly if it will not be able to create an instance of some parameter, will not it?
Otherwise we SHOULD check parameters on _null_ in EACH constructors where **@Inject** is used, don't we?

Reply:
Agree